### ISSUE TITLE
Cannot receive mail from docker host

### ISSUE THREAD

**Issue by jio3252** [tomav/docker-mailserver#1097]
## Context
While I use this docker mailserver as a general mailserver for my domain, my docker host is also running a send only exim server to send me reports for other services.
I would like to have that exim server send mails to my docker-mailserver, however the mails keep getting dropped.

The exim server can send mails to other providers just find, so there is no issue.

The first issue I could resolve by adding my (public) host ip to the mynetworks list in postfix
However the logs of the docker container tell me that postfix tries to reconnect to the host (which cannot succeed) and the mail is never delivered. 
I added the host domain (which uses a different domain name) as well as the sending mailaddress to a hashmap referenced in smtpd_recipient_restrictions, however this did not change anything.

I suspect this is connected to a check to see if the sending mail address is valid, however I cannot seem to disable this check.

## Expected Behavior
It should be possible to allow the host to send mail to the container

## Actual Behavior
Any mailserver on the host is blocked by the container mailserver

## Steps to Reproduce

1.
2.
3.
4.

## Your Environment

* Amount of RAM available: 6GB
* Mailserver version used: newest
* Docker version used:  newest
* Environment settings relevant to the config:
* Any relevant stack traces ("Full trace" preferred):

**Comment by johansmitsnl** [comment]
Use exim to deliver emails to localhost:25 as a relay.

**Comment by jio3252** [comment]
Thank you for the reply.

However the setup where docker-mailserver acts as a relay won't work either.

My setup:

On the host, exim4 is configured to use a smarthost. 
The hostname exim uses is localhost and it is listening to port 127.0.0.1:25
The target relay is docker-mailserver

Docker mailserver uses the name domain.tld and is available at 172.17.0.10 as well as the public IP.
The IP of the hosts is already in my_networks in the postfix config.

So far everything should work.

The mail (to test@domain.tld) is successfully sent to docker-mailserver and then sent to root with the note "address unrouteable". 

When sending mails from the host to root@domain.tld it succeeds, presumably because root exists on both mailsystems.

when I use another hostname for the exim server, mail delivery fails completely as before.

It also does not seem to make a difference wheter I use the public ip as the relay or the internal docker ip of docker-mailserver


Any ideas?

**Comment by johansmitsnl** [comment]
You should add an alias for the root@domain.tld email address to an existing account. Or create a normal root@domain.tld account

**Comment by jio3252** [comment]
Semms like I spoke too soon.

This only works when sending mail via the mail command to root.

Sending to other users in docker-mailserver returns with "Unrouteable address"

When sending to root from other services like logwatch, it also fails and the return address also says "root@localhost" and not "root@domian.tld" like it does with the mail command.

I don't understand why the setup behaves this way at all.
Very confusing.

**Comment by johansmitsnl** [comment]
Then add the email address with `root@localhost` as an account

**Comment by jio3252** [comment]
adding root@localhost (or root@domain.tld) as an accound does not seem to make a difference.

I am still getting 
```
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a permanent error. The following address(es) failed:

  root@localhost
```

and when sending to other users "unrouteable address"


I have build similar setups before (without docker) and never had this kind of issue. Am I overlooking something basic? Maybe I need to change the hostname of the docker host? (already changed the hostname in exim)

**Comment by johansmitsnl** [comment]
This piece of log is from the docket server of from Exim?

**Comment by jio3252** [comment]
The mail also included:

`(ultimately generated from root@domain.tld) `

Meaning its from the postfix server, since exim uses localhost as a domain

**Comment by jio3252** [comment]
I got one step further by changing the hostname of the actual host. Since it was the same name as the domain exim got confused. This was probably a stupid and obvious mistake.

I can now send simple mail to any user in docker-mailserver, however I cannot send reports as I indendet to as the exim logs include this new error:

```
2019-01-26 13:02:02 1gnMet-00014j-Eh <= root@dockerhost U=root P=local S=84277
2019-01-26 13:02:03 1gnMet-00014j-Eh **user@domain.tld R=smarthost T=remote_smtp_smarthost: message is too big (transport limit = 1)
2019-01-26 13:02:03 1gnMex-0001BZ-8q <= <> R=1gnMet-00014j-Eh U=Debian-exim P=local S=1671
2019-01-26 13:02:03 1gnMet-00014j-Eh Completed
2019-01-26 13:02:03 1gnMex-0001BZ-8q H=172.21.0.25 [172.21.0.25] Connection refused
2019-01-26 13:02:03 1gnMex-0001BZ-8q == root@dockerhost R=hub_user_smarthost T=remote_smtp_smarthost defer (111): Connection refused
```

I do not see any error about this in the postfix logs, but the way I read it the message is refused by the smarthost, which is docker-mailserver.

**Comment by johansmitsnl** [comment]
I don't know exim but to me 2 items:
* remote_smtp_smarthost: message is too big
* Nothing is listening on port (25 I guess) on IP 172.17.0.10

On 172.17.0.10 is this the docker container?

**Comment by johansmitsnl** [comment]
If you use a normal mail client you can send email? The default limit is big enough.
Can you `telnet 172.17.0.10 25` to the docker server?

**Comment by jio3252** [comment]
When I use thunderbird to send a text mail, it works fine.

When I use thunderbird to send a mail with a pdf attached, I get:

`The size of the message you are trying to send exceeds the global size limit (10240000 bytes) of the server. The message was not sent; reduce the message size and try again`

Exim is not involved in this

**Comment by johansmitsnl** [comment]
That is a large pdf and a different warning then before.

**Comment by jio3252** [comment]
You are right, though it was 1 MB but it comploain about the mail being bigger then 10.
I suspected the transport limit = 1 sets it to 1 MB.

How can I change the global size limit in docker-mailserver? Or similar settings about the size

**Comment by jio3252** [comment]
After increasing POSTFIX_MESSAGE_SIZE_LIMIT I can sent mails with large attachments in thunderbird.

Semms the only remaining issue is with exim

**Comment by erik-wramner** [comment]
Is this still a problem? If it is, do you think there is a problem with docker-mailserver? I'd like to close the issue unless there is something we need to do about it.

If you did find a solution after Jan 27 please post it so that other users can see the solution.

**Comment by erik-wramner** [comment]
Closing due to inactivity, please reopen or file a new issue if the problem persists.
